tests/kernel/protection/src/main.c

```diff
@@ -0,0 +1,180 @@
+/*
+ * Parts derived from tests/kernel/fatal/src/main.c, which has the
+ * following copyright and license:
+ *
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <tc_util.h>
```

Comment:
tc_util is still being in use?

I thought ztest.h was superseding it. It's nicer and easier to use. Could you move towards that one instead?